modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/idp/v1/IdPSuccessTest.java

```diff
@@ -277,6 +280,85 @@ public void testAddIdP() throws IOException {
         assertNotNull(idPId);
     }
 
+    @Test()
+    public void addIdPWithoutAuthenticator() throws IOException {
+
+        String body = readResource("add-idp-without-authenticator.json");
+        Response response = getResponseOfPost(IDP_API_BASE_PATH, body);
+        response.then()
+                .log().ifValidationFails()
+                .assertThat()
+                .statusCode(HttpStatus.SC_CREATED)
+                .body("federatedAuthenticators.authenticators", Matchers.emptyIterable())
+                .header(HttpHeaders.LOCATION, notNullValue());
+
+        String location = response.getHeader(HttpHeaders.LOCATION);
+        assertNotNull(location);
+        String idpIdWithoutAuth = location.substring(location.lastIndexOf("/") + 1);
+        assertNotNull(idpIdWithoutAuth);
+
+        deleteCreatedIdP(idpIdWithoutAuth);
+    }
+
+    // TODO: add a comment
```

Comment:
Attend to these TODOs

Reply:
This is updated with be797d1079d847b27cc76440a6cb21ecb3476a5d